src/poll/awakener.rs

```diff
@@ -0,0 +1,104 @@
+use std::io;
+
+use crate::event::EventedId;
+use crate::poll::Poller;
+use crate::sys;
+
+/// Awakener allows cross-thread waking of a `Poller` instance.
+///
+/// When created it will cause events with
+/// [`Ready::READABLE`](crate::event::Ready::READABLE) and the provided `id` if
+/// [`wake`](Awakener::wake) is called, possibly from another thread.
+///
+/// # Notes
+///
+/// Only a single `Awakener` should active per `Poller` instance, the `Awakener`
+/// can be cloned using [`try_clone`](Awakener::try_clone) if more are needed.
+/// What happens if multiple `Awakener`s are registered with the same `Poller`
+/// instance is undefined.
+///
+/// Awakener should be [`drain`]ed after its awoken a number of times, to not
+/// block the waking side. See [`drain`] for more information.
+///
+/// [`drain`]: Awakener::drain
+///
+/// # Examples
+///
+/// Wake a `Poller` instance from another thread.
+///
+/// ```
+/// # fn main() -> Result<(), Box<std::error::Error>> {
+/// use std::thread;
+/// use std::time::Duration;
+///
+/// use mio_st::event::{Events, EventedId, Ready};
+/// use mio_st::net::TcpStream;
+/// use mio_st::poll::{Poller, Awakener};
+///
+/// let mut poller = Poller::new()?;
+/// let mut events = Events::new();
+///
+/// let wake_id = EventedId(10);
+/// let awakener = Awakener::new(&mut poller, wake_id)?;
+///
+/// let handle = thread::spawn(move || {
+///     // Working hard, or hardly working?
+///     thread::sleep(Duration::from_millis(500));
```

Comment:
The same issue could appear here, if the system is under load and this thread exits before the actual polling. (This might explain the travis-ci failure). ¯\_(ツ)_/¯